docs/metrics/averaging-methods.md

```diff
@@ -0,0 +1,121 @@
+---
+subtitle: Macro, Micro, Weighted
+---
+
+# Averaging Methods
+
+For multiclass workflows like classification or object detection, metrics such as precision, recall, and F1-score are
+computed **per class**. To compute a single value that represents model performance across all classes, these per-class
+scores need to be aggregated. There are a few different averaging methods for doing this, most notably:
+
+- [**Macro**](#macro-average): unweighted mean of all per-class scores
+- [**Micro**](#micro-average): global average of per-sample TP, FP, FN scores
+- [**Weighted**](#weighted-average): mean of all per-class scores, weighted by sample sizes for each class
+
+## Example: Multiclass Classification
+
+Let’s consider the following multiclass classification metrics, computed across a total of 10 samples:
+
+| Class | # Samples | # True Positives  | # False Positives | # False Negatives | Precision | Recall | F1-score |
+| --- | --- | --- | --- | --- | --- | --- | --- |
+| `Airplane` | 3 | 2 | 1 | 1 | 0.67 | 0.67 | 0.67 |
+| `Boat` | 1 | 1 | 3 | 0 | 0.25 | 1.0 | 0.4 |
+| `Car` | 6 | 3 | 0 | 3 | 1.0 | 0.5 | 0.67 |
+| Total | 10 | 6 | 4 | 4 | - | - | - |
+
+
+### Macro Average
+
+**Macro average** is perhaps the most straightforward among the numerous options and is computed by taking an
+**unweighted** mean of all the per-class scores:
+
+$$
+\begin{align}
+\text{F1}_\text{macro} &= \frac{\text{F1}_\texttt{Airplane} + \text{F1}_\texttt{Boat} + \text{F1}_\texttt{Car}}{3} \\[1em]
+&= \frac{0.67 + 0.4 + 0.67}{3} \\[1em]
+&= 0.58
+\end{align}
+$$
+
+### Micro Average
+
+In contrast to macro, **micro average** computes a **global** average by counting the sums of true positive (TP), false
+negative (FN) and false positive (FP).
+
+**Micro precision** and **micro recall** are computed with the standard precision and recall formulas, using the total
+TP/FP/FN counts across all classes:
+
+<div class="grid" markdown>
+$$
+\begin{align}
+\text{Precision}_\text{micro} &= \frac{\text{TP}_\text{Total}}{\text{TP}_\text{Total} + \text{FP}_\text{Total}} \\[1em]
+&= \frac{6}{6 + 4} \\[1em]
+&= 0.6
+\end{align}
+$$
+
+$$
+\begin{align}
+\text{Recall}_\text{micro} &= \frac{\text{TP}_\text{Total}}{\text{TP}_\text{Total} + \text{FN}_\text{Total}} \\[1em]
+&= \frac{6}{6 + 4} \\[1em]
+&= 0.6
+\end{align}
+$$
+</div>
+
+What about **micro F1**? Plug the micro-averaged values for precision and recall into the standard formula for F1 score:
+
+$$
+\begin{align}
+\text{F1}_\text{micro} &= 2 \times \frac{\text{Precision}_\text{micro} \times \text{Recall}_\text{micro}}{\text{Precision}_\text{micro} + \text{Recall}_\text{micro}} \\[1em]
+&= 2 \times \frac{0.6 * 0.6}{0.6 + 0.6} \\[1em]
```

Comment:
nit: can we use \times here instead of *?

Reply:
Updated :+1: